armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -473,6 +472,25 @@ def _conserveComponentDensity(self, b, oldHeight, oldVolume):
                 for key in c.getNuclides():
                     c.setNumberDensity(key, c.getNumberDensity(key) / growth)
 
+    def _resolveMisalignment(self, b, blkBelow):
+        c = self.expansionData.getTargetComponent(b)
+        cBelow = self.expansionData.getTargetComponent(blkBelow)
+        cIdx = _getCompIndexForPreferredComp(c)
+        cBelowIdx = _getCompIndexForPreferredComp(cBelow)
+        if cIdx > cBelowIdx:
+            blkBelow.p.ztop = c.zbottom
+            blkBelow.p.height = blkBelow.p.ztop - blkBelow.p.zbottom
+        else:
+            b.p.zbottom = cBelow.ztop
```

Comment:
Can you provide a short explanation of what the bug was and how this fixed it?

I'm specifically curious about the use of the component index for the target components in each block, and what exactly that's determining about the blocks.